src/net/mqtt_control.rs

```diff
@@ -249,5 +265,30 @@ pub fn save_settings(
 
     channel.context_mut().save_configuration();
 
+    let settings = channel.context().settings();
+
+    let mut buf = [0u8; 256];
+    for (ch_indices, _) in ChannelSettings::iter_indices() {
```

Comment:
@jordens This is all very hacky currently, but I'm struggling to figure out how to do the following:
1. Get a partial path to `booster.channel[x]`
2. Iterate over the paths within `booster.channel[x]`
3. Combine (1) and (2) to save the final path to settings

I _think_ all of this code roughly does this, but it's not immediately clear. I would have expected that Miniconf would give me something like:
`booster.channels[x].path_to::<BoosterSettings>()` or similar. Any ideas here?

Reply:
1. Partial key lookup (path to indices) should be working (You'll get `TooShort(depth)`)
2. Subtree iteration in miniconf is https://github.com/quartiq/miniconf/issues/132
It's pretty clear and natural to do that there. It's a bit of work but sounds like it'll slot in just fine.
3. Rejoice

I don't see how `Tree::path_to::<T>()` would work. What if there are multiple `T` fields?

Reply:
I'm fine with hardcoding `/booster/channel/<x>` for `save <x>` here.